lib/web_ui/lib/src/engine/semantics/focusable.dart

```diff
@@ -34,6 +34,24 @@ class Focusable extends RoleManager {
 
   final AccessibilityFocusManager _focusManager;
 
+  /// Requests focus as a result of a route (e.g. dialog) deciding that the node
+  /// managed by this class should be focused by default when nothing requests
+  /// focus explicitly.
+  ///
+  /// This method of taking focus is different from the regular method of using
+  /// the [SemanticsObject.hasFocus] flag, as in this case the framework is not
+  /// explicitly request focus. Instead, the DOM element is being focus directly
+  /// programmatically, simulating the screen reader choosing a default element
+  /// to focus on.
```

Comment:
Should we inform the framework about the element we chose here? That way, when the user hits tab, the framework would continue the traversal correctly.

Reply:
We do inform the framework via a `didGainAccessibilityFocus` event, which happen by relevant nodes receiving the DOM `focus` event.